src/dag-link.js

```diff
@@ -5,25 +5,52 @@ const mh = require('multihashes')
 // Link represents an IPFS Merkle DAG Link between Nodes.
 class DAGLink {
   constructor (name, size, multihash) {
-    this.name = name
-    this.size = size
+    this._name = name
+    this._size = size
+    this._multihash = null
 
     if (typeof multihash === 'string') {
-      this.multihash = mh.fromB58String(multihash)
+      this._multihash = mh.fromB58String(multihash)
     } else if (Buffer.isBuffer(multihash)) {
-      this.multihash = multihash
+      this._multihash = multihash
     }
+  }
+
+  toString () {
+    const mhStr = mh.toB58String(this.multihash)
+    return `DAGLink <${mhStr} - name: "${this.name}", size: ${this.size}>`
+  }
 
-    this.json = {
+  toJSON () {
+    return {
       name: this.name,
       size: this.size,
-      hash: this.multihash ? mh.toB58String(this.multihash) : undefined
+      hash: this.multihash ? mh.toB58String(this.multihash) : null
```

Comment:
No, they're not. But let's not get into that now, will change it back to undefined.